kivy/uix/behaviors/resize.py

```diff
@@ -0,0 +1,383 @@
+    '''
+
+    resizing_right = BooleanProperty(False)
+    '''A State which is enabled/disabled depending on the position relative to the right resize border
+    It is switched to True when mouse is inside the right resize border and False when it isn't.
+    It adjusts the mouse cursor and manages resizing when touch is moved.
+
+    :attr:`resizing_right` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    resizing_up = BooleanProperty(False)
+    '''A State which is enabled/disabled depending on the position relative to the upper resize border
+    It is switched to True when mouse is inside the upper resize border and False when it isn't.
+    It adjusts the mouse cursor and manages resizing when touch is moved.
+
+    :attr:`resizing_up` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    resizing_down = BooleanProperty(False)
+    '''A State which is enabled/disabled depending on the position relative to the lower resize border
+    It is switched to True when mouse is inside the lower resize border and False when it isn't.
+    It adjusts the mouse cursor and manages resizing when touch is moved.
+
+    :attr:`resizing_down` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    resizing = BooleanProperty(False)
+    '''State of widget resizing.
+    It is switched to True when a resize border is touched and back to False when it is released.
+    It manages resizing when touch is moved.
+
+    :attr:`resizing` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    dont_move = BooleanProperty(False)
+    '''Move widget when resizing down or left.
+    Actual position has to keep position on screen when widget has a FloatLayout parent
+    and is being resized.
+    Resizing and changing position variables is problematic inside movement restricting widgets,
+    (StackLayout, BoxLayout, others) this property disables that.
+
+    :attr:`dont_move` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to False.
+    '''
+
+    def __init__(self, **kwargs):
+        super(ResizableBehavior, self).__init__(**kwargs)
+        Window.bind(mouse_pos=lambda obj, val: self.on_mouse_move(val))
+        self.cursor = None
+        self.oldpos = []
+        self.oldsize = []
+        self.cursor = ResizableCursor(parent=self, size_hint=(None, None), pos_hint=(None, None))
+
+    def on_enter(self):
+        self.on_enter_resizable()
+
+    def on_leave(self):
+        self.cursor.hidden = True
+
+    def on_enter_resizable(self):
+        Window.show_cursor = False
+        self.cursor.hidden = False
+
+    def on_leave_resizable(self):
+        Window.show_cursor = True
+        self.cursor.hidden = True
+
+    def on_mouse_move(self, pos):
+        if self.hovering:
+            if self.cursor:
+                self.cursor.on_mouse_move(pos)
+            if not self.resizing:
```

Comment:
this one could be put into upper if (and you'll spare 4 spaces) + it'll look cleaner